app/models/atmosphere/deployment.rb

```diff
@@ -10,10 +10,24 @@
 # Deployments provide a m:n link between appliances and virtual machines.
 module Atmosphere
   class Deployment < ActiveRecord::Base
+    extend Enumerize
     belongs_to :appliance,
       class_name: 'Atmosphere::Appliance'
 
     belongs_to :virtual_machine,
       class_name: 'Atmosphere::VirtualMachine'
+
+    validates :billing_state, presence: true
+
+    enumerize :billing_state, in: ["initial", "prepaid", "expired", "error"], predicates: true
+
+    after_create :initial_billing
```

Comment:
No because I only want to run initial billing if the create process returns success - otherwise the appliance creation process should never proceed to the billing stage.